File: bankingtool.client/src/app/dto/request/bank-account/transfer-amount-request.ts
export interface TransferAmountRequestObject {
  request: TransferAmountRequest;
}

export interface TransferAmountRequest {
  account: TransferAmountRequestAccount;
  transaction: TransferAmountRequestTransaction;
}

export interface TransferAmountRequestAccount {
  fromAccountId: number;
  toAccountId: number;
}

export interface TransferAmountRequestTransaction {
  amount: number;
  description: string;
}

export const getTransferAmountRequestObject = (fromAccountId: number, toAccountId: number, amount: number, description: string): TransferAmountRequestObject => {
  let model: TransferAmountRequestObject = {
    request: {
      account: {
        toAccountId: toAccountId,
        fromAccountId: fromAccountId,
      },
      transaction: {
        amount: amount,
        description: description
      }
    }
  }
  return model;
}

File: BankingTool.Model/Dto/RequestDto/BankAccount/TransferAmountRequest.cs
namespace BankingTool.Model.Dto.RequestDto.BankAccount
{
    public class TransferAmountRequestObject
    {
        public TransferAmountRequest Request { get; set; }
    }
    public class TransferAmountRequest
    {
        public TransferAmountRequestAccount Account { get; set; }
        public TransferAmountRequestTransaction Transaction { get; set; }
    }
    public class TransferAmountRequestAccount
    {
        public int FromAccountId { get; set; }
        public int ToAccountId { get; set; }
    }
    public class TransferAmountRequestTransaction
    {
        public int Amount { get; set; }
        public string Description { get; set; }
    }
}
